src/accelerate/checkpointing.py

```diff
@@ -214,3 +230,322 @@ def load_custom_state(obj, path, index: int = 0):
+    is_main_process: bool = True,
+    state_dict: Optional[dict] = None,
+    max_shard_size: Union[int, str] = "10GB",
+    save_function: Callable = torch.save,
+    safe_serialization: bool = False,
+    variant: Optional[str] = None,
+    keys_to_ignore_on_save: Optional[List[str]] = None,
+):
+    """
+    Save a model and its configuration file to a directory, so that it can be re-loaded using load_checkpoint_in_model
+
+    Arguments:
+        save_directory (`str` or `os.PathLike`):
+            Directory to which to save. Will be created if it doesn't exist.
+        is_main_process (`bool`, *optional*, defaults to `True`):
+            Whether the process calling this is the main process or not. Useful when in distributed training like TPUs
+            and need to call this function on all processes. In this case, set `is_main_process=True` only on the main
+            process to avoid race conditions.
+        state_dict (nested dictionary of `torch.Tensor`):
+            The state dictionary of the model to save. Will default to `self.state_dict()`, but can be used to only
+            save parts of the model or if special precautions need to be taken when recovering the state dictionary of
+            a model (like when using model parallelism).
+        save_function (`Callable`):
+            The function to use to save the state dictionary. Useful on distributed training like TPUs when one need to
+            replace `torch.save` by another method.
+        max_shard_size (`int` or `str`, *optional*, defaults to `"10GB"`):
+            The maximum size for a checkpoint before being sharded. Checkpoints shard will then be each of size lower
+            than this size. If expressed as a string, needs to be digits followed by a unit (like `"5MB"`).
+
+            <Tip warning={true}>
+
+            If a single weight of the model is bigger than `max_shard_size`, it will be in its own checkpoint shard
+            which will be bigger than `max_shard_size`.
+
+            </Tip>
+
+        safe_serialization (`bool`, *optional*, defaults to `False`):
+            Whether to save the model using `safetensors` or the traditional PyTorch way (that uses `pickle`).
+        variant (`str`, *optional*):
+            If specified, weights are saved in the format pytorch_model.<variant>.bin.
+        keys_to_ignore_on_save (`List[str]`, *optional*):
+            The list of key that should not be saved
+    """
+
+    if safe_serialization and not is_safetensors_available():
+        raise ImportError("`safe_serialization` requires the `safetensors library: `pip install safetensors`.")
+
+    if os.path.isfile(save_directory):
+        logger_simple.error(f"Provided path ({save_directory}) should be a directory, not a file")
+        return
+
+    os.makedirs(save_directory, exist_ok=True)
+
+    # Only save the model itself if we are using distributed training
+    model_to_save = unwrap_model(model)
```

Comment:
Should be removed. The user should do
```
save_model(accelerator.unwrap_model(model))
```

Reply:
Thx for the fix. I've added that in the description of the function. See fix [8d853b4](https://github.com/huggingface/accelerate/pull/1641/commits/8d853b4c7918360b91877737c363b5722b4c8e48)